### ISSUE TITLE
Docling serve at scale

### ISSUE THREAD

**Issue by aniketmaurya** [docling-project/docling-serve#10]
Docling is a great project! Got to know about this from [Spacy-layout](https://github.com/explosion/spacy-layout).

This is powered by vanilla FastAPI, which is good but won't scale and lacks stuff like dynamic batching and autoscaling. I would suggest to use a library specialized for serving ML based APIs like [LitServe](https://github.com/lightning-ai/litserve) or RayServe.

**Comment by gsogol** [comment]
Also, to add to the question above regarding scaling, how do you scale this based hundreds of requests per second? If you're running in the cloud, do you spin up multiple containers?

**Comment by aniketmaurya** [comment]
**Single container:** So the benchmark shows BERT-Large model with automatic batching and multiprocessing. A single model process can runs prediction on a batch of 16-32 requests to increase the throughput. Additionally, if GPU memory allows it can also spin up extra process to handle more requests. The requests are load balanced on process level via uvicorn socket. 

And yes, in cloud you can also spin up multiple containers for further scale.

**Comment by vishnoianil** [comment]
> **Single container:** So the benchmark shows BERT-Large model with automatic batching and multiprocessing. A single model process can runs prediction on a batch of 16-32 requests to increase the throughput. Additionally, if GPU memory allows it can also spin up extra process to handle more requests. The requests are load balanced on process level via uvicorn socket.
> 
> And yes, in cloud you can also spin up multiple containers for further scale.

Agree, we need both the options to scale the apis. Thanks for raising the issue and capturing the details.

**Comment by aniketmaurya** [comment]
Thank you @vishnoianil! If the maintainers are okay with this then I can send a PR for this.

**Comment by sir3mat** [comment]
Are there any k8s manifest or helm chart for docling api service?

**Comment by flefevre** [comment]
It will be definitely good to have helm chart for your beautiful project

**Comment by dolfim-ibm** [comment]
We are working on an Operator
